.github/workflows/sphinx_docs.yml

```
@@ -0,0 +1,57 @@
+name: sphinx docs
+
+on:
+  push: # run on push to main and PR
+    branches:
+      - main
+    pull_request:
+
+env:
+  # Python version used to build sphinx docs
+  DOCS_PYTHON_VERSION: "3.12"
+
+jobs:
+  docs:
+    name: Sphinx documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.DOCS_PYTHON_VERSION }}
+      
+      # base the Python cache on the hash of all pyproject.toml,
+      # which includes Python requirements.
+      # if any change, the cache is invalidated.
+      - name: Cache pip
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('pyproject.toml') }}
+          restore-keys: |
+            pip-${{ hashFiles('pyproject.toml') }}
+            pip-
+
+      # TODO: Is installing the dev environment the right call?
+      # Install dependencies
+      - name: Install dependencies
+        run: pip install -e ".[dev]"
```

It would be better if we didn't have to install all the dev dependencies; I'm not sure if package + doc optional dependencies is enough, if we have documentation for python code that imports optional dependencies.  Maybe worth trying to see what fails?

```suggestion
      # Install package + dependencies for building docs
      - name: Install dependencies
        run: pip install -e ".[doc]"
```